src/librustc_trans/back/msvc/mod.rs

```diff
@@ -46,283 +46,179 @@ pub fn link_exe_cmd(sess: &Session) -> Command {
-        if major <= 7 {
-            // In Windows SDK 7.x, x86 libraries are directly in the Lib folder,
-            // x64 libraries are inside, and it's not necessary to link against
-            // the SDK 7.x when targeting ARM or other architectures.
-            let x86 = match &sess.target.target.arch[..] {
-                "x86" => true,
-                "x86_64" => false,
-                _ => return None,
-            };
-            Some(if x86 {path} else {path.join("x64")})
-        } else if major <= 8 {
-            // Windows SDK 8.x installs libraries in a folder whose names
-            // depend on the version of the OS you're targeting. By default
-            // choose the newest, which usually corresponds to the version of
-            // the OS you've installed the SDK on.
-            let extra = match windows_sdk_v8_subdir(sess) {
-                Some(e) => e,
-                None => return None,
-            };
-            ["winv6.3", "win8", "win7"].iter().map(|p| path.join(p)).find(|part| {
-                fs::metadata(part).is_ok()
-            }).map(|path| {
-                path.join("um").join(extra)
-            })
-        } else if let Some(mut ver) = ver {
-            // Windows SDK 10 splits the libraries into architectures the same
-            // as Windows SDK 8.x, except for the addition of arm64.
-            // Additionally, the SDK 10 is split by Windows 10 build numbers
-            // rather than the OS version like the SDK 8.x does.
-            let extra = match windows_sdk_v10_subdir(sess) {
-                Some(e) => e,
-                None => return None,
-            };
-            // To get the correct directory we need to get the Windows SDK 10
-            // version, and so far it looks like the "ProductVersion" of the SDK
-            // corresponds to the folder name that the libraries are located in
-            // except that the folder contains an extra ".0". For now just
-            // append a ".0" to look for find the directory we're in. This logic
-            // will likely want to be refactored one day.
-            ver.push(".0");
-            let p = path.join(ver).join("um").join(extra);
-            fs::metadata(&p).ok().map(|_| p)
-        } else { None }
+    fn get_sdk8_dir() -> Option<PathBuf> {
+        LOCAL_MACHINE.open(r"SOFTWARE\Microsoft\Microsoft SDKs\Windows\v8.0".as_ref()).ok()
+        .and_then(|key| {
+            key.query_str("InstallationFolder").ok()
+        }).map(|root| {
+            Path::new(&root).join("Lib").join("Win8")
+        })
     }
 
-    fn windows_sdk_v8_subdir(sess: &Session) -> Option<&'static str> {
-        match &sess.target.target.arch[..] {
-            "x86" => Some("x86"),
-            "x86_64" => Some("x64"),
-            "arm" => Some("arm"),
-            _ => return None,
+    // We choose the linker toolchain that has the same host as rustc.
+    // This allows cross compiling to actually work.
```

Comment:
Mainly just to avoid having to detect bitness of the host OS. I'll add a FIXME comment saying this should probably detect the host OS bitness.